rio_tiler/io/stac.py

```diff
@@ -147,14 +136,30 @@ class STACReader(BaseReader):
 
     """
 
-    filepath: str
-    item: Optional[Dict] = None
-    minzoom: int = 0
-    maxzoom: int = 30
-    include_assets: Optional[Set[str]] = None
-    exclude_assets: Optional[Set[str]] = None
-    include_asset_types: Set[str] = field(default_factory=lambda: DEFAULT_VALID_TYPE)
-    exclude_asset_types: Optional[Set[str]] = None
+    def __init__(
+        self,
+        filepath: str,
+        item: Optional[Dict] = None,
+        minzoom: int = 0,
+        maxzoom: int = 30,
+        include_assets: Optional[Set[str]] = None,
+        exclude_assets: Optional[Set[str]] = None,
+        include_asset_types: Set[str] = None,
+        exclude_asset_types: Optional[Set[str]] = None,
+        reader: Type[BaseReader] = COGReader,
+        **kwargs: Any,
+    ):
+        """Create STACReader instance."""
+        self.filepath = filepath
+        self.item = item
+        self.minzoom = minzoom
+        self.maxzoom = maxzoom
+        self.include_assets = include_assets
+        self.exclude_assets = exclude_assets
+        self.include_asset_types = include_asset_types or DEFAULT_VALID_TYPE
+        self.exclude_asset_types = exclude_asset_types
+        self.reader = reader
+        self.reader_options = kwargs
```

Comment:
Because we need to define some variable (filepath) and have other with default value we have to have a `__init__` here.